giraph-core/src/test/java/org/apache/giraph/comm/RequestTest.java

```diff
@@ -67,251 +71,280 @@
+        }
+        vertexIdMutations.put(new IntWritable(i), mutations);
       }
-      vertexIdMutations.put(new IntWritable(i), mutations);
-    }
 
-    // Send the request
-    SendPartitionMutationsRequest<IntWritable, IntWritable, IntWritable>
+      // Send the request
+      SendPartitionMutationsRequest<IntWritable, IntWritable, IntWritable>
         request = new SendPartitionMutationsRequest<IntWritable, IntWritable,
         IntWritable>(partitionId,
         vertexIdMutations);
-    GiraphMetrics.init(conf);
-    client.sendWritableRequest(workerInfo.getTaskId(), request);
-    client.waitAllRequests();
+      GiraphMetrics.init(conf);
+      client.sendWritableRequest(workerInfo.getTaskId(), request);
+      client.waitAllRequests();
 
-    // Stop the service
-    client.stop();
-    server.stop();
+      // Stop the service
+      client.stop();
+      server.stop();
 
-    // Check the output
-    ConcurrentMap<IntWritable,
+      // Check the output
+      ConcurrentMap<IntWritable,
         VertexMutations<IntWritable, IntWritable, IntWritable>>
         inVertexIdMutations =
         serverData.getPartitionMutations().get(partitionId);
-    int keySum = 0;
-    for (Entry<IntWritable,
+      int keySum = 0;
+      for (Entry<IntWritable,
         VertexMutations<IntWritable, IntWritable, IntWritable>> entry :
         inVertexIdMutations
-        .entrySet()) {
-      synchronized (entry.getValue()) {
-        keySum += entry.getKey().get();
-        int vertexValueSum = 0;
-        for (Vertex<IntWritable, IntWritable, IntWritable> vertex : entry
+          .entrySet()) {
+        synchronized (entry.getValue()) {
+          keySum += entry.getKey().get();
+          int vertexValueSum = 0;
+          for (Vertex<IntWritable, IntWritable, IntWritable> vertex : entry
             .getValue().getAddedVertexList()) {
-          vertexValueSum += vertex.getValue().get();
-        }
-        assertEquals(3, vertexValueSum);
-        assertEquals(2, entry.getValue().getRemovedVertexCount());
-        int removeEdgeValueSum = 0;
-        for (Edge<IntWritable, IntWritable> edge : entry.getValue()
+            vertexValueSum += vertex.getValue().get();
+          }
+          assertEquals(3, vertexValueSum);
+          assertEquals(2, entry.getValue().getRemovedVertexCount());
+          int removeEdgeValueSum = 0;
+          for (Edge<IntWritable, IntWritable> edge : entry.getValue()
             .getAddedEdgeList()) {
-          removeEdgeValueSum += edge.getValue().get();
+            removeEdgeValueSum += edge.getValue().get();
+          }
+          assertEquals(20, removeEdgeValueSum);
+        }
+      }
+      assertEquals(55, keySum);
+    }
+  }
+
+  /**
+   * Tests that we run for different input parameters.
+   */
+  @RunWith(Parameterized.class)
+  public static class ParameterizedTest extends SharedSetup {
+
+    @Parameterized.Parameter(value = 0)
+    public int numVertices;
+
```

Comment:
Looks like if you only need a single parameter, you can just use:
@Parameters
public static Object[] data() {
    return new Object[] { "first test", "second test" };
}

https://github.com/junit-team/junit4/wiki/parameterized-tests

Reply:
I'll simplify this then, thanks!